app/models/queue_item.rb

```diff
@@ -5,11 +5,27 @@ class QueueItem < ActiveRecord::Base
   delegate :category, to: :video
   delegate :title, to: :video, prefix: :video
 
+  validates_numericality_of :position, {only_integer: true}
+
   def rating
-    review = Review.where(user: user, video: video).first
     review.rating if review
   end
 
+  def rating=(new_rating)
+    new_rating = nil if new_rating == ""
+
+    if review
+      review.update_column(:rating, new_rating)
+    elsif new_rating
+        review = Review.new(user: user, video: video, rating: new_rating)
```

Comment:
These two lines are indented too far. I'm glad to see this code worked for you :smiley: